docs/modules/preprocessing/dim_reduction/fpca.rst

```diff
@@ -0,0 +1,30 @@
+Functional Principal Component Analysis (FPCA)
+==============================================
+
+This module provides tools to analyse functional data using FPCA. FPCA is
+a common tool used to reduce dimensionality while preserving the maximum
+quantity of variance in the data. FPCA be applied to a functional data object
+in either a basis representation or a discretized representation. The output
+of FPCA are orthogonal functions (usually a much smaller sample than the input
+data sample) that represent the most important modes of variation in the
+original data sample.
+
+For a detailed example please view :ref:`sphx_glr_auto_examples_plot_fpca.py`,
+where the process is applied to several datasets in both discretized and basis
+forms.
+
```

Comment:
The problem with that approach is that too much functionality will be included in the same class, and the fit method will become too large. It may be very bad for future maintenance. 
Second, there are attributes that are exclusively for FPCABasis and attributes that are exclusively for FPCADiscretized. 
Having too much functionality in the same class may also confuse the users about what attributes to use when they want to apply it to a FDadaGrid compared to when they want to apply it for a FDataBasis.
Lastly, regularization is only available for FDataBasis. Which is another attribute that distinguishes both estimators.

Reply:
Im still not convinced, but we can always change it later, if necessary.